src/movement/control_system.cpp

```diff
@@ -187,265 +195,355 @@ void ControlSystem::InputControlMessage(robosub::control msg)
+    current_state.yaw_left_integral = current_integral[5] / _PI_OVER_180;
+
+    current_state.forward_derivative = current_derivative[0];
+    current_state.strafe_left_derivative = current_derivative[1];
+    current_state.dive_derivative = current_derivative[2];
+    current_state.roll_right_derivative = current_derivative[3] / _PI_OVER_180;
+    current_state.pitch_down_derivative = current_derivative[4] / _PI_OVER_180;
+    current_state.yaw_left_derivative = current_derivative[5] / _PI_OVER_180;
+
+    current_state.forward_rotation_control = rotation_control[0];
+    current_state.strafe_left_rotation_control = rotation_control[1];
+    current_state.dive_rotation_control = rotation_control[2];
+    current_state.roll_right_rotation_control = rotation_control[3] / _PI_OVER_180;
+    current_state.pitch_down_rotation_control = rotation_control[4] / _PI_OVER_180;
+    current_state.yaw_left_rotation_control = rotation_control[5] / _PI_OVER_180;
+
+    current_state.forward_translation_control = translation_control[0];
+    current_state.strafe_left_translation_control = translation_control[1];
+    current_state.dive_translation_control = translation_control[2];
+    current_state.roll_right_translation_control = translation_control[3] / _PI_OVER_180;
+    current_state.pitch_down_translation_control = translation_control[4] / _PI_OVER_180;
+    current_state.yaw_left_translation_control = translation_control[5] / _PI_OVER_180;
 
     return current_state;
 }
-void ControlSystem::PublishThrusterMessage()
-{
-    pub->publish(tp);
-}
 
-/*
-string ControlSystem::statestring(control_packet::State state)
+std::string ControlSystem::state_to_string(uint8_t state)
 {
-    string s;
     switch (state)
     {
-        case robosub::control::STATE_NONE:     s = "NONE"; break;
-        case robosub::control::STATE_ABSOLUTE: s = "ABSOLUTE"; break;
-        case robosub::control::STATE_RELATIVE: s = "RELATIVE"; break;
-        case robosub::control::STATE_ERROR:    s = "ERROR"; break;
-        default: s = "UNKNOWN"; break;
+        case robosub::control::STATE_NONE:
+            return "No State";
+            break;
+        case robosub::control::STATE_ERROR:
+            return "Error State";
+            break;
+        case robosub::control::STATE_ABSOLUTE:
+            return "Absolute State";
+            break;
+        case robosub::control::STATE_RELATIVE:
+            return "Relative State";
+            break;
+        default:
+            return "Unknown State";
+            break;
     }
+    return "Unknown State";
+}
 
-    return s;
+/**
+ * Publishes a thruster command.
+ *
+ * @return None.
+ */
+void ControlSystem::PublishThrusterMessage()
+{
+    pub->publish(tp);
 }
-*/
 
-VectorXd ControlSystem::motor_control(Vector12d state)
+/**
+ * Calculates the motor control signal.
+ *
+ * @return The control signal C to send to the thrusters.
+ */
+VectorXd ControlSystem::calculate_motor_control()
```

Comment:
I would actually argue this should all stay together as it's doing one single thing. It's a lot of stuff, but it should all be here to ensure everything is called in the correct order.